app/src/main/java/ch/sdp/vibester/helper/BuzzerGameManager.kt

```diff
@@ -22,7 +23,15 @@ class BuzzerGameManager: GameManager() {
      * Play current song with media player.
      */
     fun playSong() {
-        mediaPlayer = AudioPlayer.playAudio(currentSong.getPreviewUrl())
+        if(hasInternet) {
+            mediaPlayer = AudioPlayer.playAudio(currentSong.getPreviewUrl())
+        } else {
+            mediaPlayer = AudioPlayer.playAudio(currentSong.getPreviewUrl())
+            /* TODO: OFFLINE
+            var media = File(context.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS), "extract_of_$currentSong.getTrackName()")
+            mediaPlayer = AudioPlayer.playAudio(media.absolutePath)
+             */
+        }

```

Comment:
Could you just put these modifications into GameManager instead? I deleted BuzzerGameManager when refactoring and put all the equivalent functions into GameManager instead so you should find it there

Reply:
Done on local, will push soon once I make sure the merge was successful